kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs

```diff
@@ -327,3 +309,65 @@ fn print_report<'tcx>(ctx: &GotocCtx, tcx: TyCtxt<'tcx>) {
         tcx.sess.warn(&msg);
     }
 }
+
+fn codegen_results(
+    tcx: TyCtxt,
+    rustc_metadata: EncodedMetadata,
+    machine: &MachineModel,
+) -> Box<dyn Any> {
+    let work_products = FxHashMap::<WorkProductId, WorkProduct>::default();
+    Box::new((
+        CodegenResults {
+            modules: vec![],
+            allocator_module: None,
+            metadata_module: None,
+            metadata: rustc_metadata,
+            crate_info: CrateInfo::new(tcx, machine.architecture.clone()),
+        },
+        work_products,
+    ))
+}
+
+/// Collect all harnesses in the current crate.
+fn collect_harnesses<'tcx>(tcx: TyCtxt<'tcx>, gcx: &GotocCtx) -> Vec<MonoItem<'tcx>> {
+    // Filter proof harnesses.
+    tcx.hir_crate_items(())
+        .items()
+        .filter_map(|hir_id| {
+            let def_id = hir_id.def_id.to_def_id();
+            gcx.is_proof_harness(def_id).then(|| MonoItem::Fn(Instance::mono(tcx, def_id)))
+        })
+        .collect()
+}
+
+/// Retrieve all items that need to be processed.
+fn codegen_items<'tcx>(tcx: TyCtxt<'tcx>, gcx: &GotocCtx) -> Vec<MonoItem<'tcx>> {
```

Comment:
Rename to `get_codegen_items`? The current name suggests it performs codegen, which it doesn't.

Reply:
That's a good point.